src/states/elder/mod.rs

```diff
@@ -714,24 +714,30 @@ impl Elder {
             msg.routing_message()
         );
 
-        let dst = msg.message_dst();
-        let signed_msg = msg.signed_routing_message();
-        if self.in_location(dst) {
-            self.check_signed_message_trust(signed_msg)?;
-            self.check_signed_message_integrity(signed_msg)?;
-            self.update_our_knowledge(signed_msg);
-
-            if dst.is_multiple() {
-                // Broadcast to the rest of the section.
-                if let Err(error) = self.send_signed_message(&msg) {
-                    debug!("{} Failed to send {:?}: {:?}", self, signed_msg, error);
-                }
+        if msg.message_dst().is_multiple() {
+            // Relay closer to the destination or broadcast to the rest of our section.
+            if let Err(error) = self.send_signed_message(&msg) {
```

Comment:
does noy seem to handle when we are a hop toward the single node destination.

Reply:
Yes, I just realized this by seeing a whole bunch of failing tests :grimacing: